pydantic_mongo/fields.py

```
@@ -1,9 +1,71 @@
+            ValueError: If the value doesn't match any enum values
+        """
+        try:
+            # Try to find the enum by value
+            return enum_type(value)
+        except ValueError:
+            # If we get here, no match was found
+            valid_values = [e.value for e in enum_type]
+            raise ValueError(
+                f"Invalid value '{value}' for {enum_type.__name__}. "
+                f"Valid values: {valid_values}"
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)